### PR TITLE
Add guestConfig metadata doc for wasm middleware

### LINKED ISSUE #3617: Docs for http wasm middleware guest config
**What content needs to be created or modified?**
Add `guestConfig`  for wasm middleware component

**The associated pull request from dapr/dapr, dapr/components-contrib, or other Dapr code repos**

https://github.com/dapr/components-contrib/pull/2918

### PR DESCRIPTION
**Pull request by Taction**
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [x] [Read the contribution guide](https://docs.dapr.io/contributing/docs-contrib/contributing-docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

Add guestConfig metadata doc for wasm middleware.

## Issue reference

Please reference the issue this PR will close: #3617

### CODE REVIEW

File: daprdocs/content/en/reference/components-reference/supported-middleware/middleware-wasm.md [hunk 1]
```diff
@@ -45,6 +47,7 @@ How to compile this is described later.
 | Field | Details                                                        | Required | Example        |
 |-------|----------------------------------------------------------------|----------|----------------|
 | url   | The URL of the resource including the Wasm binary to instantiate. The supported schemes include `file://`. The path of a `file://` URL is relative to the Dapr process unless it begins with `/`. | true     | `file://hello.wasm` |
+| guestConfig   | An optional configuration passed to WASM guests. Users can pass an arbitrary string to be parsed by the guest code. | false     | `enviroment=production`,`{"environment":"production"}` |
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-middleware/middleware-wasm.md]:
```suggestion
| guestConfig   | An optional configuration passed to Wasm guests. Users can pass an arbitrary string to be parsed by the guest code. | false     | `enviroment=production`,`{"environment":"production"}` |
```

File: daprdocs/content/en/reference/components-reference/supported-middleware/middleware-wasm.md [hunk 2]
```diff
@@ -116,6 +119,49 @@ If using TinyGo, compile as shown below and set the spec metadata field named
 tinygo build -o router.wasm -scheduler=none --no-debug -target=wasi router.go`
 ```
 
+### wasm config example
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-middleware/middleware-wasm.md]:
```suggestion
### Wasm `guestConfig` example
```

File: daprdocs/content/en/reference/components-reference/supported-middleware/middleware-wasm.md [hunk 3]
```diff
@@ -116,6 +119,49 @@ If using TinyGo, compile as shown below and set the spec metadata field named
...
+
+Here is an example of how to use `guestConfig` to pass configurations to wasm. In wasm code, you can use the function `handler.Host.GetConfig` defined in guest sdk to get the configuration. In the following example, the wasm middleware parses the executed `environment` from json config defined in the component.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-middleware/middleware-wasm.md]:
```suggestion
Here is an example of how to use `guestConfig` to pass configurations to Wasm. In Wasm code, you can use the function `handler.Host.GetConfig` defined in guest SDK to get the configuration. In the following example, the Wasm middleware parses the executed `environment` from JSON config defined in the component.
```

**Review by hhunter-ms — CHANGES_REQUESTED** [review]
quick review

**Comment by msfussell** [comment]
@Taction - Can you take a look at these few updates? Thx

**Comment by Taction** [comment]
Thanks for the review and the remind. Fixed.

**Review by msfussell — APPROVED** [review]
LGTM

**Review by hhunter-ms — APPROVED** [review]

**Pull request merged**